### PR TITLE
Use spf13/cobra for docker update

### PR DESCRIPTION
**Pull request by yongtang**
This fix is part of the effort to convert commands to spf13/cobra #23211.

Thif fix coverted command `docker update` to use spf13/cobra

Signed-off-by: Yong Tang yong.tang.github@outlook.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
LGTM

**Comment by thaJeztah** [comment]
sorry @yongtang, this one needs a rebase as well :cry:

**Comment by yongtang** [comment]
Thanks @thaJeztah. The PR has been rebased.

File: api/client/container/update.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+	var restartPolicy containertypes.RestartPolicy
+	if opts.restartPolicy != "" {
+		restartPolicy, err = runconfigopts.ParseRestartPolicy(opts.restartPolicy)
+		if err != nil {
+			return err
+		}
+	}
+
+	resources := containertypes.Resources{
```
Comment (vdemeester) [inline api/client/container/update.go]: I feel the pointer should stay the same :angel:

Comment (yongtang) [inline api/client/container/update.go]: Thanks @vdemeester. Are you referring to changing the fields of `updateOptions` to pointers and always prefix with `fl` (like old version) as below?

```
type updateOptions struct {
	flblkioWeight       *uint16
 	flCPUPeriod         *int64
 	flCPUQuota          *int64
        ...
 }
```

Comment (vdemeester) [inline api/client/container/update.go]: on the `Resources` struct :angel:

Comment (yongtang) [inline api/client/container/update.go]: Change
```
Memory:            memory,
```
to
```
Memory:            flMemory,
```
?

Comment (vdemeester) [inline api/client/container/update.go]:
```go
CPUShares: *opts.cpuShares,
```

Comment (yongtang) [inline api/client/container/update.go]: I see. Let me update the pull request.

**Comment by icecrime** [comment]
What the?? On gccgo:

```
06:25:33 FAIL: docker_cli_help_test.go:15: TestHelpTextVerify.pN52_github_com_docker_docker_integration_cli.DockerSuite
06:25:33 
06:25:33 docker_cli_help_test.go:143:
06:25:33     c.Fatal(err)
06:25:33 ... Error: Error on "logout" help. non-empty stderr:"Aborted\n\npanic during panic\n\nruntime stack:\nstack trace unavailable\n"
```

File: api/client/container/update.go (hunk 1 above)
Comment (yongtang) [inline api/client/container/update.go]: Thanks @vdemeester the pointers has been changed and PR has been updated. Let me know if there are any issues.

**Comment by yongtang** [comment]
Thanks @icecrime for the review. I am hoping the failure is a one-time failure but I will keep an eye on it in case it is repeatable.

**Comment by thaJeztah** [comment]
ping @vdemeester PTAL

File: api/client/container/update.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/cli"
+	runconfigopts "github.com/docker/docker/runconfig/opts"
+	containertypes "github.com/docker/engine-api/types/container"
+	"github.com/docker/go-units"
+	"github.com/spf13/cobra"
+)
+
+type updateOptions struct {
+	blkioWeight       *uint16
```
Comment (vdemeester) [inline api/client/container/update.go]: I don't think pointer are needed there, just in the resource struct :angel:

Comment (yongtang) [inline api/client/container/update.go]: Thanks @vdemeester but if we don't use pointers like
```
blkioWeight       *uint16
...
cpuShares         *int64
```
here, we may not be able to reference pointers in `resource` 
```
		BlkioWeight:       *opts.blkioWeight,
....
 		CPUShares:         *opts.cpuShares,
```
as well?

Maybe I missed something?

Comment (LK4D4) [inline api/client/container/update.go]: @vdemeester ping :)

**Comment by thaJeztah** [comment]
Sorry @yongtang, this needs a rebase now :cry:

**Comment by yongtang** [comment]
Thanks @thaJeztah the pull request has been rebased.

**Comment by yongtang** [comment]
Ping @vdemeester @thaJeztah.

File: api/client/container/update.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.containers = args
+			opts.nFlag = cmd.Flags().NFlag()
+			return runUpdate(dockerCli, &opts)
+		},
+	}
+	cmd.SetFlagErrorFunc(flagErrorFunc)
+
+	flags := cmd.Flags()
+	opts.blkioWeight = flags.Uint16("blkio-weight", 0, "Block IO (relative weight), between 10 and 1000")
```
Comment (LK4D4) [inline api/client/container/update.go]: Use `flags.Uint16Var` to avoid pointers. Take `api/client/container/run.go` as reference. Thanks!

**Comment by LK4D4** [comment]
@yongtang Sorry for delay! Let me know if you want us to carry PR.

**Comment by yongtang** [comment]
@LK4D4 Thanks for the review. I updated the pull request to change the `blkioWeight` in opts. Please let me know if there are any issues.

File: api/client/container/update.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.containers = args
+			opts.nFlag = cmd.Flags().NFlag()
+			return runUpdate(dockerCli, &opts)
+		},
+	}
+	cmd.SetFlagErrorFunc(flagErrorFunc)
+
+	flags := cmd.Flags()
+	flags.Uint16Var(&opts.blkioWeight, "blkio-weight", 0, "Block IO (relative weight), between 10 and 1000")
```
Comment (LK4D4) [inline api/client/container/update.go]: Please use `Var` stuff for all options.

Comment (yongtang) [inline api/client/container/update.go]: Thanks @LK4D4 let me update the PR shortly.

**Comment by yongtang** [comment]
Thanks @LK4D4 The pull request has been updated. Please let me know if there are any other issues.

**Comment by LK4D4** [comment]
@yongtang perfect! I'll merge if ci pass.

**Comment by vdemeester** [comment]
LGTM 🐮 assuming it's green 😇

**Pull request merged**